core/src/main/java/com/linecorp/armeria/client/retry/RetryStrategy.java

```diff
@@ -112,13 +146,30 @@ static RetryStrategy onServerErrorStatus(Backoff backoff) {
      *
      * @param backoffFunction A {@link BiFunction} that returns the {@link Backoff} or {@code null} (no retry)
      *                        according to the given {@link HttpStatus} and {@link Throwable}
+     *
+     * @deprecated Use {@link RetryRuleBuilder#onStatus(Predicate)},
+     *             {@link RetryRuleBuilder#onException(Predicate)} and
+     *             {@link RetryRuleBuilder#thenBackoff(Backoff)}.
+     *             For example:
+     *             <pre>{@code
+     *             RetryRule.onStatus(HttpStatus.TOO_MANY_REQUESTS)
+     *                      .onException(ex -> ex instanceof ClosedStreamException)
+     *                      .thenBackoff(myBackoff);
+     *             }</pre>
      */
+    @Deprecated
     static RetryStrategy onStatus(
             BiFunction<? super HttpStatus, ? super Throwable, ? extends Backoff> backoffFunction) {
-        // TODO(trustin): Apply a different backoff for UnprocessedRequestException.
         return new HttpStatusBasedRetryStrategy(backoffFunction);
     }
 
+    /**
+     * Converts a {@link RetryStrategy} to a {@link RetryRule}.
+     */
+    default RetryRule toRetryRule() {
```

Comment:
Instead of adding this to a deprecated type, just convert with an internal helper in `RetryingClientBuilder`